### ISSUE TITLE
Selections on source breaks when src.rpm has Provides

### ISSUE THREAD

**Issue by ignatenkobrain** [openSUSE/libsolv#329]
```
repo test 0 testtags <inline>
#>=Pkg: a 1 1 src
#>=Req: x
#>=Pkg: b 1 1 src
#>=Prv: y
system x86_64 rpm
poolflags implicitobsoleteusescolors
solverflags keepexplicitobsoletes
job noop selection a-1-1.src canon
job noop selection b-1-1.src canon
result jobs <inline>
#>job noop name (a . src) = 1-1 [setevr,setarch]
#>job noop name (b . src) = 1-1 [setevr,setarch]
```

→

```
Results differ:
-job noop name (b . src) = 1-1 [setevr,setarch]
```

**Comment by ignatenkobrain** [comment]
src.rpms which use DynamicBuildRequires from RPM 4.15 put `Provides: rpmlib(DynamicBuildRequires)` so that it is easy to identify such sourcerpms.

**Comment by Conan-Kudo** [comment]
Why is it in the `Provides` instead of just reading a `Requires` for it?

**Comment by ignatenkobrain** [comment]
@Conan-Kudo Requires are used to indicate that SRPM uses that feature while Provides are used to indicate that SRPM contains generated BuildRequires.

**Comment by Conan-Kudo** [comment]
@ignatenkobrain Yes, but the Provides thing doesn't make sense at all. If it has the Requires for it, the of course it means it needs the feature to be supported, and should trigger the desired behavior...

**Comment by ignatenkobrain** [comment]
@Conan-Kudo well, whether it makes sense or not, it breaks selections... Which is libsolv problem. Provides are not prohibited for solvables with ARCH_SRC. For RPM, open a ticket on RPM issue tracker.

**Comment by mlschroe** [comment]
Bah, that design makes no sense at all. Who came up with that?

Ok, I'll make libsolv more resistant against such provides.

**Comment by Conan-Kudo** [comment]
@mlschroe Please come up with a better way and talk to rpm upstream about it. As you can tell, I don't like this either...

**Comment by ignatenkobrain** [comment]
@mlschroe that was @ffesti's idea :) I have just implemented it.

**Comment by mlschroe** [comment]
ok, fixed with commit 69439fe4eea683200aa6449ac6632f4283997e49
